From da33828c013848cdb612360bdcf8436d1ff635a9 Mon Sep 17 00:00:00 2001
Message-Id: <da33828c013848cdb612360bdcf8436d1ff635a9.1430208948.git.chang-joon.lee@intel.com>
From: Andrzej Datczuk <andrzej.datczuk@intel.com>
Date: Thu, 23 Apr 2015 21:05:42 +0200
Subject: [PATCH] REVERTME [VPG]: drm/i915: Perfmon RC6 WA BB PIPE_CONTROL
 bits refined

Added PIPE_CONTROL_WRITE_FLUSH(12) and PIPE_CONTROL_GLOBAL_GTT_IVB(24)
bits when sending PIPE_CONTROL commad before and after
NOA programming commands in RC6 WA BB. This is done in
i915_perfmon_update_workaround_bb function.

This change is needed to correctly program all NOA registers in
case of using perfmon with RC6 enabled. Without it, configuration
sent to DFT Ring through 0x9888 proxy register on RC6 exit is
ignored due to yet blocked DFT - timing issues.

Patch is marked with REVERTME as Perfmon IOCTL will be removed
once OA couters are exposed via 'perf event' framework.

Issue: GMINL-3120
Change-Id: I70a7d5e9ecf1f6973f40dd4d511b04d0d8b58fba
Signed-off-by: Andrzej Datczuk <andrzej.datczuk@intel.com>
---
 drivers/gpu/drm/i915/i915_perfmon.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_perfmon.c b/drivers/gpu/drm/i915/i915_perfmon.c
index 96d7c3f..df5cae2 100644
--- a/drivers/gpu/drm/i915/i915_perfmon.c
+++ b/drivers/gpu/drm/i915/i915_perfmon.c
@@ -578,7 +578,8 @@ static void *emit_load_register_imm(void *mem, __u32 reg, __u32 val)
 static void *emit_cs_stall_pipe_control(void *mem)
 {
 	mem = emit_dword(mem, GFX_OP_PIPE_CONTROL(6));
-	mem = emit_dword(mem, PIPE_CONTROL_CS_STALL);
+	mem = emit_dword(mem, PIPE_CONTROL_CS_STALL|PIPE_CONTROL_WRITE_FLUSH|
+			      PIPE_CONTROL_GLOBAL_GTT_IVB);
 	mem = emit_dword(mem, 0);
 	mem = emit_dword(mem, 0);
 	mem = emit_dword(mem, 0);
@@ -619,7 +620,7 @@ int i915_perfmon_update_workaround_bb(struct drm_i915_private *dev_priv,
 		return 0;
 	}
 
-	/* diable RC6 WA BB */
+	/* disable RC6 WA BB */
 	I915_WRITE(GEN8_RC6_WA_BB, 0x0);
 
 	buffer_tail = dev_priv->rc6_wa_bb.address;
-- 
1.7.9.5

